app/services/profile_fields/remove.rb

```diff
@@ -0,0 +1,32 @@
+# This service removes a profile field and ensures that the corresponding store
+# accessor also gets removed immediately.
+module ProfileFields
+  class Remove
+    def self.call(id)
+      new(id).call
+    end
+
+    attr_reader :profile_field, :error_message
+
+    def initialize(id)
+      @id = id
+      @success = false
+    end
+
+    def call
+      @profile_field = ProfileField.find(@id)
+      if @profile_field.destroy
+        accessor = profile_field.attribute_name.to_s
+        Profile.undef_method(accessor) if accessor.in?(Profile.stored_attributes[:data])
```

Comment:
**Question**: TIL about undef_method. Do we call it instead of remove_method because we don't want it to search superclasses and mixed-in modules? and undef_method, no matter where it exists, will just not respond to this accessor.

Reply:
Oh, no particular reason. `undef_method` was the one I remembered. Remove method is semantically more correct though, so I'll change it. Thanks!

Reply:
Hm, changing this actually made the existing specs fails. Since I'm not concerned about mixings or superclasses here, I'll just change it back to `undef_method`. Just FYI @Ridhwana.